docs/development.rst

```diff
@@ -24,15 +24,28 @@ Publishing
 
 Publishing to `PyPI <https://pypi.org/project/pyhf/>`__ and `TestPyPI <https://test.pypi.org/project/pyhf/>`__
 is automated through the `PyPA's PyPI publish GitHub Action <https://github.com/pypa/gh-action-pypi-publish>`__.
-To publish a release to PyPI one simply needs to run
+Publishing a release to PyPI begins with creating a new branch :code:`release/cut-<release tag>`
+and then from that branch running
 
 .. code-block:: console
 
     bumpversion [major|minor|patch]
 
-to update the release version and get a tagged commit and then push the commit
-and tag to :code:`master` with
+to update the release version and get a tagged commit.
+Then, push only the branch to GitHub and open a PR to ensure that the code
+you are going to publish is indeed stable.
+Once the PR has passed all checks and has been approved, a maintainer should merge
+the branch from the command line
 
 .. code-block:: console
 
-    git push origin master --tags
+    git checkout master
+    git merge release/cut-<release tag>
+
+and then push the commit and tag to :code:`master` with
+
+.. code-block:: console
+
+    git push origin master <release tag>
+
+which will both merge and close the PR and start the release publication workflow.
```

Comment:
The tag push may generate a separate `push` event (additionally to branch's `push` and `pull_request`). Maybe it's worth mentioning that one can visually see more jobs than they expect.

Reply:
Correct. As can be seen with release `v0.2.2`

```
git push origin master v0.2.2
```

[kicked off 5 actions](https://github.com/scikit-hep/pyhf/commit/2175c9ff9120791a628e2164a05107310894c5b0/checks): 2 publish actions (TestPyPI and PyPI), 2 `push` actions, and then our Binder trigger action. So good point on this.

Though I think that all of this is gong to get rewritten given PR #650 as @kratsg is changing it from Issue driven to PR driven.